src/hotspot/cpu/x86/x86_64.ad

```diff
@@ -9153,6 +9166,22 @@ instruct andI2L_rReg_imm65535(rRegL dst, rRegI src, immI_65535 mask)
   ins_pipe(ialu_reg);
 %}
 
+// Can skip int2long conversions after AND with small bitmask
+instruct convI2LAndI_reg_immIbitmask(rRegL dst, rRegI src, immI_bitmask mask, rRegI tmp, rFlagsReg cr)
+%{
+  predicate(VM_Version::supports_bmi2());
+  ins_cost(125);
+  effect(TEMP tmp, KILL cr);
+  match(Set dst (ConvI2L (AndI src mask)));
+  format %{ "bzhiq $dst, $src, $mask \t# using $tmp as TEMP, int & immI_bitmask -> long" %}
+  ins_encode %{
+    assert(VM_Version::supports_bmi2(), "required");
```

Comment:
The assert is redundant here. The predicate already guarantees that invariant.

Reply:
Agreed.